.github/workflows/run_and_deploy.yml

```diff
@@ -0,0 +1,83 @@
+name: Run and Deploy
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+  # Allow manual runs through the web UI
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check_style:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+      - name: Install dependencies
+        run: |
+          python -m pip install tox
+      - name: Check Style
+        run: |
+          tox -e codestyle
+
+  build_docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+      - name: Install dependencies
+        run: |
+          python -m pip install tox
+      - name: Build Docs
+        run: |
+          tox -e py311
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: files
+          path: |
+            /home/runner/work/infrastructure/infrastructure/docs/_build/html/*
+
+  deploy:
+    needs: [build_docs]
+    if: |
+       (
+         github.event_name != 'pull_request' && (
+         github.ref_name == 'main' ||
+         github.event_name == 'workflow_dispatch'
+         )
+       )
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+    - name: Get current date
+      id: date
+      run: echo "::set-output name=date::$(date +'%Y%m%d')"
+    - name: Download all workflow run artifacts
+      uses: actions/download-artifact@v3
+      with:
+        name: files
+        path: ~/files
+    - name: Display structure of downloaded files
+      run: ls -R
+      working-directory: /home/runner/files
+    - name: Deploy to GitHub Pages
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.infra_automation }}
```

Comment:
This uses a key that I used from my own account, in future this will need to be created by a numfocus org admin.